Source/Core/Core/ReliableUDPManager/ReliableUDPConnection.h

```diff
@@ -0,0 +1,122 @@
+// Copyright 2013 Dolphin Emulator Project
+// Licensed under GPLv2
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <SFML/Network.hpp>
+#include "Common/CommonTypes.h"
+#include "Common/Timer.h"
+#include <queue>
+#include <map>
+#include <memory>
+#include "Common/Thread.h"
+
+class ReliableUDPConnection
+{
+#define udpBitType	u32
+#define udpBitMax	UINT32_MAX
+
+public:
+	ReliableUDPConnection(std::shared_ptr<sf::UdpSocket> socket, sf::IpAddress adr, u16 port);
+	~ReliableUDPConnection();
+	
+	void StoreSend(sf::Packet& packet);
+	sf::Socket::Status Send(bool sendAck = true);
+	sf::Socket::Status SendUnreliable(sf::Packet& packet);
+
+	bool Receive(sf::Packet& packet);
+	bool GrabMessage(sf::Packet& packet);
+	
+	bool CheckIfAlive();
+	void Disconnect();
+
+	void ClearBuffers();
+	
+	sf::IpAddress GetAddress(){ return m_remoteAddress; };
+	unsigned short  GetPort(){ return m_remotePort; };
+
+	int AmountToSend(){ return (int) m_toBeSent.size();};
+	bool isReadyToSend(){ return (!m_toBeSent.empty() || !m_resend.empty()); }
```

Comment:
Upper-case the function name here and add a space before the opening brace (this goes for all functions above too)